src/signed_data.rs

```diff
@@ -243,36 +245,95 @@ impl<'a> SubjectPublicKeyInfo<'a> {
     }
 }
 
-/// A signature algorithm.
-pub struct SignatureAlgorithm {
+/// A detail-less error when a signature is not valid.
+#[derive(Debug, Copy, Clone)]
+pub struct InvalidSignature;
```

Comment:
I'm not sure I understand why this exists separate from the `Error` enum if it's also exported. Can you expand on your thinking here for my edification?

Reply:
Ah, I think I understand now having read further: if we added a variant to `Error` then the trait impl of verification could return any of the variants, including more detail, when what we really want is a generic undetailed indication of failure.